cilium-dbg/cmd/bpf_metrics_list_test.go

```diff
@@ -98,19 +94,18 @@ func (s *BPFMetricsMapSuite) TestDumpMetrics(c *C) {
 		},
 	}
 
-	rawDump := dumpAndRead(metricsMap, func(maps []*mockmaps.MetricsMockMap, args ...interface{}) {
+	rawDump := dumpAndRead(t, metricsMap, func(maps []*mockmaps.MetricsMockMap, args ...interface{}) {
 		for _, m := range maps {
 			listMetrics(m)
 		}
-	}, c)
+	})
 
 	var got jsonMetrics
 	err := json.Unmarshal([]byte(rawDump), &got)
-	c.Assert(err, IsNil, Commentf("invalid JSON output: '%s', '%s'", err, rawDump))
+	require.NoError(t, err, "invalid JSON output: '%s', '%s'", err, rawDump)
 
 	sort.Slice(got, func(i, j int) bool {
 		return got[i].Packets <= got[j].Packets
 	})
-
-	c.Assert(want, checker.DeepEquals, got)
+	require.Equal(t, len(want), len(got))
```

Comment:
Shouldn't we compare the slices' content here as well?

Reply:
thanks, it's addressed in https://github.com/cilium/cilium/pull/32210